projects/addon-mobile/directives/ripple/ripple.directive.ts

```diff
@@ -1,79 +1,158 @@
-
-                    return race(
-                        timer(TOUCH_MOVE_DELAY).pipe(map(TUI_FALSE_HANDLER)),
-                        touchEnd$.pipe(map(TUI_TRUE_HANDLER)),
-                    ).pipe(
-                        take(1),
-                        // eslint-disable-next-line rxjs/no-unsafe-takeuntil
-                        takeUntil(touchMove$),
-                        tap(() => {
-                            this.renderer.setStyle(
-                                ripple,
-                                'background',
-                                this.tuiRipple || null,
-                            );
-                            this.renderer.appendChild(this.el, ripple);
+                take(1),
+                tap(() => {
+                    this.renderer.appendChild(element, ripple);
+                }),
+                switchMap(isTap => {
+                    const animation = ripple.animate(RIPPLE_ON_ANIMATION, {
+                        duration: tuiGetDuration(this.animationSpeed),
+                    });
+                    const animationEndOn$ = tuiTypedFromEvent(animation, 'finish');
+
+                    return isTap
+                        ? animationEndOn$
+                        : race(
+                              touchEnd$.pipe(switchMap(() => animationEndOn$)),
+                              animationEndOn$.pipe(switchMap(() => touchEnd$)),
+                          ).pipe(takeUntil(touchMove$), defaultIfEmpty(undefined));
+                }),
+                switchMap(() =>
+                    tuiTypedFromEvent(
+                        ripple.animate(RIPPLE_OFF_ANIMATION, {
+                            duration: tuiGetDuration(this.animationSpeed),
                         }),
-                        switchMap(isTap =>
-                            isTap
-                                ? animationEndOn$
-                                : race(
-                                      touchEnd$.pipe(switchMap(() => animationEndOn$)),
-                                      animationEndOn$.pipe(switchMap(() => touchEnd$)),
-                                  ),
-                        ),
-                        map(() => ripple),
-                    );
+                        'finish',
+                    ),
+                ),
+                tap(() => {
+                    element.removeChild(ripple);
                 }),
-                takeUntilDestroyed(),
+                takeUntilDestroyed(this.destroy$),
             )
-            .subscribe(element =>
-                this.renderer.setStyle(element, 'animationName', RIPPLE_OFF),
-            );
+            .subscribe();
+    }
+
+    private createRipple(
+        clientX: number,
+        clientY: number,
+        {width, height, top, left}: DOMRect,
+    ): HTMLElement {
+        const ripple: HTMLElement = this.renderer.createElement('div');
```

Comment:
I believe Angular team no longer promotes using renderer abstraction, let's ditch it and just use `DOCUMENT`. Even the guy responsible for Angular Universal advised against renderer use in Angular GDE slack.